core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory.java

```diff
@@ -202,16 +203,15 @@ private void updatePicker(@Nullable Status error) {
      * subchannel otherwise, return null.
      */
     @Nullable
-    private Status getAggregatedError() {
-      Status status = null;
+    private ConnectivityStateInfo getAggregatedState() {
+      ConnectivityStateInfo stateInfo = null;
       for (Subchannel subchannel : getSubchannels()) {
-        ConnectivityStateInfo stateInfo = getSubchannelStateInfoRef(subchannel).get();
+        stateInfo = getSubchannelStateInfoRef(subchannel).get();
         if (stateInfo.getState() != TRANSIENT_FAILURE) {
-          return null;
+          return stateInfo;
```

Comment:
If there is at least one `READY`, the aggregated state should be `READY`.
Otherwise, if there is at least one `CONNECTING`, the aggregated state should be `CONNECTING`.
Otherwise, if there is at least one `IDLE`, the aggregated state should be `IDLE`.